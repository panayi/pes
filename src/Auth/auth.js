import R from 'ramda';
import { createSelector } from 'reselect';
import { FIREBASE_PATH } from '../Firebase/firebase';

// ------------------------------------
// Constants
// ------------------------------------
const FIREBASE_AUTH_PATH = [...FIREBASE_PATH, 'auth'];

// ------------------------------------
// Selectors
// ------------------------------------

// firebaseSelector :: State -> Object
const firebaseAuthSelector = R.pathOr({}, FIREBASE_AUTH_PATH);

// isLoadedSelector :: State -> Maybe(Boolean)
const isLoadedSelector = createSelector(
  firebaseAuthSelector,
  R.compose(
    R.prop('isLoaded'),
    R.defaultTo({}),
  ),
);

// isInitializingSelector :: State -> Maybe(Boolean)
const isInitializingSelector = R.path([...FIREBASE_PATH, 'isInitializing']);

export const uidSelector = createSelector(
  firebaseAuthSelector,
  R.prop('uid'),
);

// hasUidSelector :: State -> Maybe(String)
const hasUidSelector = createSelector(
  uidSelector,
  R.compose(
    R.not,
    R.isNil,
  ),
);

// isAuthenticatingSelector :: State -> Boolean
export const isAuthenticatingSelector = createSelector(
  isInitializingSelector,
  isLoadedSelector,
  R.useWith(R.or, [
    R.identity,
    R.not,
  ]),
);

// isAuthenticatedSelector :: State -> Boolean
// Equals to: !isAuthenticatingSelector && hasUidSelector
export const isAuthenticatedSelector = createSelector(
  isAuthenticatingSelector,
  hasUidSelector,
  R.useWith(R.and, [
    R.not,
    R.identity,
  ]),
);

// isNotAuthenticatedSelector :: State -> Boolean
// Equals to: !isAuthenticatingSelector && !hasUidSelector
export const isNotAuthenticatedSelector = createSelector(
  isAuthenticatingSelector,
  hasUidSelector,
  R.useWith(R.and, [
    R.not,
    R.not,
  ]),
);
