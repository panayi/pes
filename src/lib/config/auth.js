export default {
  google: {
    type: 'popup',
    scopes: 'profile email',
  },
  facebook: {
    type: 'popup',
    scopes: 'public_profile,email,user_friends',
  },
};
