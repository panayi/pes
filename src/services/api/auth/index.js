export const login = credentials => (dispatch, getState, getFirebase) =>
  getFirebase().login(credentials);

export const getProviders = email => (dispatch, getState, getFirebase) =>
  getFirebase()
    .auth()
    .fetchProvidersForEmail(email);
