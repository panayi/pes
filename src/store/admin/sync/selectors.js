import * as R from 'ramda';
import { createSelector } from 'reselect';
import * as constants from './constants';

const SYNC_PATH = ['admin', 'sync'];

const statusSelector = R.path([...SYNC_PATH, 'status']);

const isStatusEqualTo = value => createSelector(
  statusSelector,
  R.equals(value),
);

export const isStatusIdleSelector = isStatusEqualTo(constants.STATUS_IDLE);

export const isStatusSucceededSelector = isStatusEqualTo(constants.STATUS_SUCCEEDED);

const syncedImageSelector = R.path([...SYNC_PATH, 'images']);

const syncedPostsSelector = R.path([...SYNC_PATH, 'posts']);

export const postsSelector = createSelector(
  syncedPostsSelector,
  syncedImageSelector,
  R.useWith(R.merge, [
    R.identity,
    R.map(count => ({ syncedImagesCount: count })),
  ]),
);
