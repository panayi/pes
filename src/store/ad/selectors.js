/* @flow */
import * as R from 'ramda';
import { createSelector } from 'reselect';
import propsSelector from 'utils/propsSelector';
import { PENDING_ADS } from 'services/connectData/types';
import { uidSelector } from '../auth/selectors';

// adImagesPathSelector :: (_, Props) -> Object | Nil
export const adImagesPathSelector = createSelector(
  R.compose(R.prop('adId'), propsSelector),
  adId => `ads/${adId}/images`,
);

// pendingAdImagesSelector :: State -> Object | Nil
export const pendingAdImagesPathSelector = createSelector(
  uidSelector,
  R.unless(R.isNil, uid => `${PENDING_ADS}/${uid}/images`),
);

// pendingAdSelector :: (_, Props) -> Ad | Nil
//   Ad = Object
export const pendingAdSelector = createSelector(
  propsSelector,
  R.prop('pendingAd'),
);
