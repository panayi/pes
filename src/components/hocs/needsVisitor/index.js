import React from 'react';
import { connectedRouterRedirect } from 'redux-auth-wrapper/history4/redirect';
import locationHelperBuilder from 'redux-auth-wrapper/history4/locationHelper';
import { Grid } from 'material-ui';
import BounceLoader from 'react-spinners';
import { isNotAuthenticatedSelector, isAuthenticatingSelector } from 'store/auth/selectors';

const locationHelper = locationHelperBuilder({});

const Loader = () => (
  <Grid
    align="center"
    justify="center"
    style={{ height: '100vh' }}
  >
    <BounceLoader
      color="#26A65B"
      size="60px"
    />
  </Grid>
);


export default (options = {}) => connectedRouterRedirect({
  // This sends the user either to the query param route if we have one,
  // or to the landing page if none is specified and the user is already logged in
  redirectPath: (state, ownProps) => locationHelper.getRedirectQueryParam(ownProps) || '/',
  // This prevents us from adding the query parameter when we send the user away from the login page
  allowRedirectBack: false,
  authenticatedSelector: isNotAuthenticatedSelector,
  authenticatingSelector: isAuthenticatingSelector,
  AuthenticatingComponent: Loader,
  ...options,
});
