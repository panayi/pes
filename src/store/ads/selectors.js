/* @flow */
import * as R from 'ramda';
import { createSelector } from 'reselect';
import propsSelector from 'utils/propsSelector';
import { PENDING_ADS } from 'services/connectData/types';
import { uidSelector, profilePropSelector } from 'store/auth/selectors';
import * as constants from './constants';

export const createAdPath = ['ad', 'create'];

export const createAdSelector = R.path(createAdPath);

export const isCreateAdIdleSelector = createSelector(
  createAdSelector,
  R.equals(constants.AD_CREATE_IDLE_STATE),
);

export const isCreateAdPendingSelector = createSelector(
  createAdSelector,
  R.equals(constants.AD_CREATE_PENDING_STATE),
);

export const isCreateAdCompletedSelector = createSelector(
  createAdSelector,
  R.equals(constants.AD_CREATE_COMPLETED_STATE),
);

// adImagesPathSelector :: Props -> Object | Nil
export const adImagesPathSelector = createSelector(
  R.compose(R.prop('adId'), propsSelector),
  adId => `ads/${adId}/images`,
);

// pendingAdImagesSelector :: State -> Object | Nil
export const pendingAdImagesPathSelector = createSelector(
  uidSelector,
  R.unless(R.isNil, uid => `${PENDING_ADS}/${uid}/images`),
);

// pendingAdSelector :: Props -> Ad | Nil
//   Ad = Object
export const pendingAdSelector = createSelector(
  propsSelector,
  R.prop('pendingAd'),
);

export const currentUserAdIdsSelector = createSelector(
  profilePropSelector(['ads']),
  R.keys,
);
