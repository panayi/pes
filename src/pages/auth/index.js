import React from 'react';
import { Route, Switch } from 'react-router-dom';
import needsVisitor from '../../auth/visibility/needsVisitor';
import Layout from '../../layout';
import Login from './login';

const Auth = () => (
  <Layout>
    <Switch>
      <Route exact path="/auth/login" component={Login} />
    </Switch>
  </Layout>
);

export default needsVisitor()(Auth);
