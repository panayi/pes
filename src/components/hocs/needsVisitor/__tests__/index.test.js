import React from 'react';
import { mount } from 'enzyme';
import * as authSelectors from 'store/auth/selectors';
import needsVisitor from '../index';

describe('[HOC] needsVisitor', () => {
  authSelectors.isNotAuthenticatedSelector = jest.fn();
  authSelectors.isAuthenticatingSelector = jest.fn();
  
  const location = {
    pathname: '/some-visitor-specific-url',
    search: '',
    hash: '',
  };

  const MyComponent = () => <div>i am a visitor</div>;
  const Hoc = needsVisitor({
    redirectPath: '/you-cant-see-visitor-page',
  })(MyComponent);
  const { component, store } = withMockStore(<Hoc location={location} />);
  
  it('should render wrappedComponent correctly', () => {
    authSelectors.isNotAuthenticatedSelector.mockReturnValueOnce(true);
    authSelectors.isAuthenticatingSelector.mockReturnValueOnce(false);
    const wrapper = mount(component);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render wrappedComponent when isNotAuthenticatedSelector = true', () => {
    authSelectors.isNotAuthenticatedSelector.mockReturnValueOnce(true);
    authSelectors.isAuthenticatingSelector.mockReturnValueOnce(false);
    const wrapper = mount(component);
    expect(wrapper.find(MyComponent).exists()).toBe(true);
  });
  
  it('should not render wrappedComponent and redirect, when isNotAuthenticatedSelector = false', () => {
    authSelectors.isNotAuthenticatedSelector.mockReturnValueOnce(false);
    authSelectors.isAuthenticatingSelector.mockReturnValueOnce(false);
    const wrapper = mount(component);
    expect(wrapper.find(MyComponent).exists()).toBe(false);
    expect(store.getActions()).toMatchSnapshot();
  });

  it('should render Loader when isAuthenticatingSelector = true', () => {
    authSelectors.isNotAuthenticatedSelector.mockReturnValueOnce(false);
    authSelectors.isAuthenticatingSelector.mockReturnValueOnce(true);
    const wrapper = mount(component);
    expect(wrapper).toMatchSnapshot();
  });

  xit('should get redirectPath from wrappedComponent props');
});
