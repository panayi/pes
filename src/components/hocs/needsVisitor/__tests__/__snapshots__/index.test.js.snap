// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[HOC] needsVisitor should not render wrappedComponent and redirect, when isNotAuthenticatedSelector = false 1`] = `
Array [
  Object {
    "payload": Object {
      "args": Array [
        Object {
          "hash": null,
          "pathname": "/you-cant-see-visitor-page",
          "search": "",
        },
      ],
      "method": "replace",
    },
    "type": "@@router/CALL_HISTORY_METHOD",
  },
]
`;

exports[`[HOC] needsVisitor should render Loader when isAuthenticatingSelector = true 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "firebase": Object {
        "INTERNAL": Object {
          "ErrorFactory": [Function],
          "Promise": [Function],
          "createFirebaseNamespace": [Function],
          "createSubscribe": [Function],
          "deepExtend": [Function],
          "extendNamespace": [Function],
          "factories": Object {
            "auth": [Function],
            "database": [Function],
          },
          "node": Object {
            "XMLHttpRequest": [Function],
            "localStorage": Storage {},
            "sessionStorage": Storage {},
          },
          "registerService": [Function],
          "removeApp": [Function],
          "useAsService": [Function],
        },
        "Promise": [Function],
        "SDK_VERSION": "4.5.0",
        "User": [Function],
        "_": Object {
          "authUid": null,
          "config": Object {
            "apiKey": "asdf",
            "authDomain": "asdf",
            "databaseURL": "ws://127.0.1:5000",
            "enableRedirectHandling": false,
            "messagingSenderId": "asdf",
            "storageBucket": "asdf",
            "userProfile": "users",
          },
          "watchers": Object {},
        },
        "__esModule": true,
        "app": [Function],
        "apps": Array [],
        "auth": [Function],
        "database": [Function],
        "default": [Circular],
        "initializeApp": [Function],
      },
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(AuthWrapper(MyComponent))
    location={
      Object {
        "hash": "",
        "pathname": "/some-visitor-specific-url",
        "search": "",
      }
    }
  >
    <AuthWrapper(MyComponent)
      dispatch={[Function]}
      isAuthenticated={false}
      isAuthenticating={true}
      location={
        Object {
          "hash": "",
          "pathname": "/some-visitor-specific-url",
          "search": "",
        }
      }
      redirectPath="/you-cant-see-visitor-page"
    >
      <Loader
        dispatch={[Function]}
        isAuthenticated={false}
        isAuthenticating={true}
        location={
          Object {
            "hash": "",
            "pathname": "/some-visitor-specific-url",
            "search": "",
          }
        }
        redirectPath="/you-cant-see-visitor-page"
      >
        <withStyles(GridWrapper)
          align="center"
          justify="center"
          style={
            Object {
              "height": "100vh",
            }
          }
        >
          <GridWrapper
            align="center"
            classes={
              Object {
                ".MuiGrid-spacing-xs-16-21 > .MuiGrid-typeItem-2": "MuiGrid-spacing-xs-16-21 > .MuiGrid-typeItem-2",
                ".MuiGrid-spacing-xs-24-22 > .MuiGrid-typeItem-2": "MuiGrid-spacing-xs-24-22 > .MuiGrid-typeItem-2",
                ".MuiGrid-spacing-xs-40-23 > .MuiGrid-typeItem-2": "MuiGrid-spacing-xs-40-23 > .MuiGrid-typeItem-2",
                ".MuiGrid-spacing-xs-8-20 > .MuiGrid-typeItem-2": "MuiGrid-spacing-xs-8-20 > .MuiGrid-typeItem-2",
                "align-content-xs-center": "MuiGrid-align-content-xs-center-11",
                "align-content-xs-flex-end": "MuiGrid-align-content-xs-flex-end-13",
                "align-content-xs-flex-start": "MuiGrid-align-content-xs-flex-start-12",
                "align-content-xs-space-around": "MuiGrid-align-content-xs-space-around-15",
                "align-content-xs-space-between": "MuiGrid-align-content-xs-space-between-14",
                "align-items-xs-baseline": "MuiGrid-align-items-xs-baseline-10",
                "align-items-xs-center": "MuiGrid-align-items-xs-center-7",
                "align-items-xs-flex-end": "MuiGrid-align-items-xs-flex-end-9",
                "align-items-xs-flex-start": "MuiGrid-align-items-xs-flex-start-8",
                "direction-xs-column": "MuiGrid-direction-xs-column-3",
                "direction-xs-column-reverse": "MuiGrid-direction-xs-column-reverse-4",
                "direction-xs-row-reverse": "MuiGrid-direction-xs-row-reverse-5",
                "grid-lg": "MuiGrid-grid-lg-63",
                "grid-lg-1": "MuiGrid-grid-lg-1-64",
                "grid-lg-10": "MuiGrid-grid-lg-10-73",
                "grid-lg-11": "MuiGrid-grid-lg-11-74",
                "grid-lg-12": "MuiGrid-grid-lg-12-75",
                "grid-lg-2": "MuiGrid-grid-lg-2-65",
                "grid-lg-3": "MuiGrid-grid-lg-3-66",
                "grid-lg-4": "MuiGrid-grid-lg-4-67",
                "grid-lg-5": "MuiGrid-grid-lg-5-68",
                "grid-lg-6": "MuiGrid-grid-lg-6-69",
                "grid-lg-7": "MuiGrid-grid-lg-7-70",
                "grid-lg-8": "MuiGrid-grid-lg-8-71",
                "grid-lg-9": "MuiGrid-grid-lg-9-72",
                "grid-md": "MuiGrid-grid-md-50",
                "grid-md-1": "MuiGrid-grid-md-1-51",
                "grid-md-10": "MuiGrid-grid-md-10-60",
                "grid-md-11": "MuiGrid-grid-md-11-61",
                "grid-md-12": "MuiGrid-grid-md-12-62",
                "grid-md-2": "MuiGrid-grid-md-2-52",
                "grid-md-3": "MuiGrid-grid-md-3-53",
                "grid-md-4": "MuiGrid-grid-md-4-54",
                "grid-md-5": "MuiGrid-grid-md-5-55",
                "grid-md-6": "MuiGrid-grid-md-6-56",
                "grid-md-7": "MuiGrid-grid-md-7-57",
                "grid-md-8": "MuiGrid-grid-md-8-58",
                "grid-md-9": "MuiGrid-grid-md-9-59",
                "grid-sm": "MuiGrid-grid-sm-37",
                "grid-sm-1": "MuiGrid-grid-sm-1-38",
                "grid-sm-10": "MuiGrid-grid-sm-10-47",
                "grid-sm-11": "MuiGrid-grid-sm-11-48",
                "grid-sm-12": "MuiGrid-grid-sm-12-49",
                "grid-sm-2": "MuiGrid-grid-sm-2-39",
                "grid-sm-3": "MuiGrid-grid-sm-3-40",
                "grid-sm-4": "MuiGrid-grid-sm-4-41",
                "grid-sm-5": "MuiGrid-grid-sm-5-42",
                "grid-sm-6": "MuiGrid-grid-sm-6-43",
                "grid-sm-7": "MuiGrid-grid-sm-7-44",
                "grid-sm-8": "MuiGrid-grid-sm-8-45",
                "grid-sm-9": "MuiGrid-grid-sm-9-46",
                "grid-xl": "MuiGrid-grid-xl-76",
                "grid-xl-1": "MuiGrid-grid-xl-1-77",
                "grid-xl-10": "MuiGrid-grid-xl-10-86",
                "grid-xl-11": "MuiGrid-grid-xl-11-87",
                "grid-xl-12": "MuiGrid-grid-xl-12-88",
                "grid-xl-2": "MuiGrid-grid-xl-2-78",
                "grid-xl-3": "MuiGrid-grid-xl-3-79",
                "grid-xl-4": "MuiGrid-grid-xl-4-80",
                "grid-xl-5": "MuiGrid-grid-xl-5-81",
                "grid-xl-6": "MuiGrid-grid-xl-6-82",
                "grid-xl-7": "MuiGrid-grid-xl-7-83",
                "grid-xl-8": "MuiGrid-grid-xl-8-84",
                "grid-xl-9": "MuiGrid-grid-xl-9-85",
                "grid-xs": "MuiGrid-grid-xs-24",
                "grid-xs-1": "MuiGrid-grid-xs-1-25",
                "grid-xs-10": "MuiGrid-grid-xs-10-34",
                "grid-xs-11": "MuiGrid-grid-xs-11-35",
                "grid-xs-12": "MuiGrid-grid-xs-12-36",
                "grid-xs-2": "MuiGrid-grid-xs-2-26",
                "grid-xs-3": "MuiGrid-grid-xs-3-27",
                "grid-xs-4": "MuiGrid-grid-xs-4-28",
                "grid-xs-5": "MuiGrid-grid-xs-5-29",
                "grid-xs-6": "MuiGrid-grid-xs-6-30",
                "grid-xs-7": "MuiGrid-grid-xs-7-31",
                "grid-xs-8": "MuiGrid-grid-xs-8-32",
                "grid-xs-9": "MuiGrid-grid-xs-9-33",
                "justify-xs-center": "MuiGrid-justify-xs-center-16",
                "justify-xs-flex-end": "MuiGrid-justify-xs-flex-end-17",
                "justify-xs-space-around": "MuiGrid-justify-xs-space-around-19",
                "justify-xs-space-between": "MuiGrid-justify-xs-space-between-18",
                "spacing-xs-16": "MuiGrid-spacing-xs-16-21",
                "spacing-xs-24": "MuiGrid-spacing-xs-24-22",
                "spacing-xs-40": "MuiGrid-spacing-xs-40-23",
                "spacing-xs-8": "MuiGrid-spacing-xs-8-20",
                "typeContainer": "MuiGrid-typeContainer-1",
                "typeItem": "MuiGrid-typeItem-2",
                "wrap-xs-nowrap": "MuiGrid-wrap-xs-nowrap-6",
              }
            }
            justify="center"
            style={
              Object {
                "height": "100vh",
              }
            }
          >
            <Grid
              align="center"
              alignContent="stretch"
              alignItems="stretch"
              classes={
                Object {
                  ".MuiGrid-spacing-xs-16-21 > .MuiGrid-typeItem-2": "MuiGrid-spacing-xs-16-21 > .MuiGrid-typeItem-2",
                  ".MuiGrid-spacing-xs-24-22 > .MuiGrid-typeItem-2": "MuiGrid-spacing-xs-24-22 > .MuiGrid-typeItem-2",
                  ".MuiGrid-spacing-xs-40-23 > .MuiGrid-typeItem-2": "MuiGrid-spacing-xs-40-23 > .MuiGrid-typeItem-2",
                  ".MuiGrid-spacing-xs-8-20 > .MuiGrid-typeItem-2": "MuiGrid-spacing-xs-8-20 > .MuiGrid-typeItem-2",
                  "align-content-xs-center": "MuiGrid-align-content-xs-center-11",
                  "align-content-xs-flex-end": "MuiGrid-align-content-xs-flex-end-13",
                  "align-content-xs-flex-start": "MuiGrid-align-content-xs-flex-start-12",
                  "align-content-xs-space-around": "MuiGrid-align-content-xs-space-around-15",
                  "align-content-xs-space-between": "MuiGrid-align-content-xs-space-between-14",
                  "align-items-xs-baseline": "MuiGrid-align-items-xs-baseline-10",
                  "align-items-xs-center": "MuiGrid-align-items-xs-center-7",
                  "align-items-xs-flex-end": "MuiGrid-align-items-xs-flex-end-9",
                  "align-items-xs-flex-start": "MuiGrid-align-items-xs-flex-start-8",
                  "direction-xs-column": "MuiGrid-direction-xs-column-3",
                  "direction-xs-column-reverse": "MuiGrid-direction-xs-column-reverse-4",
                  "direction-xs-row-reverse": "MuiGrid-direction-xs-row-reverse-5",
                  "grid-lg": "MuiGrid-grid-lg-63",
                  "grid-lg-1": "MuiGrid-grid-lg-1-64",
                  "grid-lg-10": "MuiGrid-grid-lg-10-73",
                  "grid-lg-11": "MuiGrid-grid-lg-11-74",
                  "grid-lg-12": "MuiGrid-grid-lg-12-75",
                  "grid-lg-2": "MuiGrid-grid-lg-2-65",
                  "grid-lg-3": "MuiGrid-grid-lg-3-66",
                  "grid-lg-4": "MuiGrid-grid-lg-4-67",
                  "grid-lg-5": "MuiGrid-grid-lg-5-68",
                  "grid-lg-6": "MuiGrid-grid-lg-6-69",
                  "grid-lg-7": "MuiGrid-grid-lg-7-70",
                  "grid-lg-8": "MuiGrid-grid-lg-8-71",
                  "grid-lg-9": "MuiGrid-grid-lg-9-72",
                  "grid-md": "MuiGrid-grid-md-50",
                  "grid-md-1": "MuiGrid-grid-md-1-51",
                  "grid-md-10": "MuiGrid-grid-md-10-60",
                  "grid-md-11": "MuiGrid-grid-md-11-61",
                  "grid-md-12": "MuiGrid-grid-md-12-62",
                  "grid-md-2": "MuiGrid-grid-md-2-52",
                  "grid-md-3": "MuiGrid-grid-md-3-53",
                  "grid-md-4": "MuiGrid-grid-md-4-54",
                  "grid-md-5": "MuiGrid-grid-md-5-55",
                  "grid-md-6": "MuiGrid-grid-md-6-56",
                  "grid-md-7": "MuiGrid-grid-md-7-57",
                  "grid-md-8": "MuiGrid-grid-md-8-58",
                  "grid-md-9": "MuiGrid-grid-md-9-59",
                  "grid-sm": "MuiGrid-grid-sm-37",
                  "grid-sm-1": "MuiGrid-grid-sm-1-38",
                  "grid-sm-10": "MuiGrid-grid-sm-10-47",
                  "grid-sm-11": "MuiGrid-grid-sm-11-48",
                  "grid-sm-12": "MuiGrid-grid-sm-12-49",
                  "grid-sm-2": "MuiGrid-grid-sm-2-39",
                  "grid-sm-3": "MuiGrid-grid-sm-3-40",
                  "grid-sm-4": "MuiGrid-grid-sm-4-41",
                  "grid-sm-5": "MuiGrid-grid-sm-5-42",
                  "grid-sm-6": "MuiGrid-grid-sm-6-43",
                  "grid-sm-7": "MuiGrid-grid-sm-7-44",
                  "grid-sm-8": "MuiGrid-grid-sm-8-45",
                  "grid-sm-9": "MuiGrid-grid-sm-9-46",
                  "grid-xl": "MuiGrid-grid-xl-76",
                  "grid-xl-1": "MuiGrid-grid-xl-1-77",
                  "grid-xl-10": "MuiGrid-grid-xl-10-86",
                  "grid-xl-11": "MuiGrid-grid-xl-11-87",
                  "grid-xl-12": "MuiGrid-grid-xl-12-88",
                  "grid-xl-2": "MuiGrid-grid-xl-2-78",
                  "grid-xl-3": "MuiGrid-grid-xl-3-79",
                  "grid-xl-4": "MuiGrid-grid-xl-4-80",
                  "grid-xl-5": "MuiGrid-grid-xl-5-81",
                  "grid-xl-6": "MuiGrid-grid-xl-6-82",
                  "grid-xl-7": "MuiGrid-grid-xl-7-83",
                  "grid-xl-8": "MuiGrid-grid-xl-8-84",
                  "grid-xl-9": "MuiGrid-grid-xl-9-85",
                  "grid-xs": "MuiGrid-grid-xs-24",
                  "grid-xs-1": "MuiGrid-grid-xs-1-25",
                  "grid-xs-10": "MuiGrid-grid-xs-10-34",
                  "grid-xs-11": "MuiGrid-grid-xs-11-35",
                  "grid-xs-12": "MuiGrid-grid-xs-12-36",
                  "grid-xs-2": "MuiGrid-grid-xs-2-26",
                  "grid-xs-3": "MuiGrid-grid-xs-3-27",
                  "grid-xs-4": "MuiGrid-grid-xs-4-28",
                  "grid-xs-5": "MuiGrid-grid-xs-5-29",
                  "grid-xs-6": "MuiGrid-grid-xs-6-30",
                  "grid-xs-7": "MuiGrid-grid-xs-7-31",
                  "grid-xs-8": "MuiGrid-grid-xs-8-32",
                  "grid-xs-9": "MuiGrid-grid-xs-9-33",
                  "justify-xs-center": "MuiGrid-justify-xs-center-16",
                  "justify-xs-flex-end": "MuiGrid-justify-xs-flex-end-17",
                  "justify-xs-space-around": "MuiGrid-justify-xs-space-around-19",
                  "justify-xs-space-between": "MuiGrid-justify-xs-space-between-18",
                  "spacing-xs-16": "MuiGrid-spacing-xs-16-21",
                  "spacing-xs-24": "MuiGrid-spacing-xs-24-22",
                  "spacing-xs-40": "MuiGrid-spacing-xs-40-23",
                  "spacing-xs-8": "MuiGrid-spacing-xs-8-20",
                  "typeContainer": "MuiGrid-typeContainer-1",
                  "typeItem": "MuiGrid-typeItem-2",
                  "wrap-xs-nowrap": "MuiGrid-wrap-xs-nowrap-6",
                }
              }
              component="div"
              container={false}
              direction="row"
              hidden={undefined}
              item={false}
              justify="center"
              spacing={16}
              style={
                Object {
                  "height": "100vh",
                }
              }
              wrap="wrap"
            >
              <div
                align="center"
                className="MuiGrid-justify-xs-center-16"
                style={
                  Object {
                    "height": "100vh",
                  }
                }
              >
                <Spinner
                  fadeIn="full"
                  name="circle"
                  noFadeIn={false}
                  overrideSpinnerClassName=""
                >
                  <div
                    className="sk-fade-in sk-spinner sk-circle"
                  >
                    <div
                      key="0"
                    />
                    <div
                      key="1"
                    />
                    <div
                      key="2"
                    />
                    <div
                      key="3"
                    />
                    <div
                      key="4"
                    />
                    <div
                      key="5"
                    />
                    <div
                      key="6"
                    />
                    <div
                      key="7"
                    />
                    <div
                      key="8"
                    />
                    <div
                      key="9"
                    />
                    <div
                      key="10"
                    />
                    <div
                      key="11"
                    />
                  </div>
                </Spinner>
              </div>
            </Grid>
          </GridWrapper>
        </withStyles(GridWrapper)>
      </Loader>
    </AuthWrapper(MyComponent)>
  </Connect(AuthWrapper(MyComponent))>
</Provider>
`;

exports[`[HOC] needsVisitor should render wrappedComponent correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "firebase": Object {
        "INTERNAL": Object {
          "ErrorFactory": [Function],
          "Promise": [Function],
          "createFirebaseNamespace": [Function],
          "createSubscribe": [Function],
          "deepExtend": [Function],
          "extendNamespace": [Function],
          "factories": Object {
            "auth": [Function],
            "database": [Function],
          },
          "node": Object {
            "XMLHttpRequest": [Function],
            "localStorage": Storage {},
            "sessionStorage": Storage {},
          },
          "registerService": [Function],
          "removeApp": [Function],
          "useAsService": [Function],
        },
        "Promise": [Function],
        "SDK_VERSION": "4.5.0",
        "User": [Function],
        "_": Object {
          "authUid": null,
          "config": Object {
            "apiKey": "asdf",
            "authDomain": "asdf",
            "databaseURL": "ws://127.0.1:5000",
            "enableRedirectHandling": false,
            "messagingSenderId": "asdf",
            "storageBucket": "asdf",
            "userProfile": "users",
          },
          "watchers": Object {},
        },
        "__esModule": true,
        "app": [Function],
        "apps": Array [],
        "auth": [Function],
        "database": [Function],
        "default": [Circular],
        "initializeApp": [Function],
      },
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(AuthWrapper(MyComponent))
    location={
      Object {
        "hash": "",
        "pathname": "/some-visitor-specific-url",
        "search": "",
      }
    }
  >
    <AuthWrapper(MyComponent)
      dispatch={[Function]}
      isAuthenticated={true}
      isAuthenticating={false}
      location={
        Object {
          "hash": "",
          "pathname": "/some-visitor-specific-url",
          "search": "",
        }
      }
      redirectPath="/you-cant-see-visitor-page"
    >
      <MyComponent
        dispatch={[Function]}
        isAuthenticated={true}
        isAuthenticating={false}
        location={
          Object {
            "hash": "",
            "pathname": "/some-visitor-specific-url",
            "search": "",
          }
        }
        redirectPath="/you-cant-see-visitor-page"
      >
        <div>
          i am a visitor
        </div>
      </MyComponent>
    </AuthWrapper(MyComponent)>
  </Connect(AuthWrapper(MyComponent))>
</Provider>
`;
